api.js

```diff
@@ -199,16 +221,22 @@ exports.init = function (sbot, config) {
         debug('announce post exists on main feed')
       }
 
-      // Ensure the main feed was "added" on the root meta feed
-      const [err3, added] = await run(find)(mf, (f) => f.feedpurpose === 'main')
+      // Ensure the main feed was "added" on the path root/v1/:shard/main
+      const [err3, v1Feed] = await run(findOrCreateV1)(mf)
       if (err3) return release(cb, err3)
+      const d = { feedpurpose: 'main' }
+      const [err4, shardFeed] = await run(findOrCreateShard)(v1Feed, mf, d)
+      if (err4) return release(cb, err4)
+      const visit = (f) => f.feedpurpose === 'main'
+      const [err5, added] = await run(find)(shardFeed, visit)
+      if (err5) return release(cb, err5)
       if (!added) {
-        debug('adding main feed to root meta feed')
-        const opts = optsForAddExisting(mf.keys, 'main', config.keys)
+        debug('adding main feed to a shard metafeed')
+        const opts = optsForAddExisting(shardFeed.keys, 'main', config.keys)
```

Comment:
:fire: / :hot_pepper: 

If I understand this right, this is announcing the location of the feed structure on the classic `main` feed, right?
If that's the case I would like to recommend that we still announce the `rootMF`, as it's much more important to know than some shard feed right?

It might be that that's not really how `metafeed/add/existing` works, but we should perhaps consider include the rootMF as meta-data then...

Reply:
I think you understood it the other way around. What this line does is that it announces the main feed as a subfeed of the shard, by publishing a message on the shard.

Reply:
Oh I thought that would just be a meta/add/existing on the sharf feed which points to a  main feed.

I'm assuming there is a message published on the main feed that points back to the root feed (to support people migrating)

Reply:
Yes there is, check the code block above this block.